Properly assign and later cleanup the connection object.

imikejackson opened the pull request:
This takes into account that multiple filters can be assigned to a command.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Properly assign and later cleanup the connection object.

This takes into account that multiple filters can be assigned to a command.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

mmarineBlueQuartz commented on Source/SVWidgetsLib/Widgets/util/AddFilterCommand.cpp:
```diff
@@ -301,9 +302,13 @@ void AddFilterCommand::disconnectFilterSignalsSlots(const AbstractFilter::Pointe
+  for(auto& connection : m_Connections)
+  {
+    QObject::disconnect(connection);
+  }
```
This method is called per filter so having a map to the filter might make more sense here, but m_Connections needs to be cleared after disconnecting the connection.  Otherwise, m_Connections will continue to grow larger every time undo() redo() are called.

mmarineBlueQuartz replied:
This is duplicated in RemoveFilterCommand.

imikejackson committed:
Clear the vector holding the connection objects after disconnecting them

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

mmarineBlueQuartz reviewed (approved):
Looks good to me.

imikejackson commented:
It works, but just by chance. I'll try a map<Filter, Connection> instead.

imikejackson committed:
Use a map to hold the QObject connection items that need to be cleaned up

This allows the proper connection object to be cleaned up when the filter is
removed.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Clean up of code: const correctness changes, remove debug output code

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>
